### DISCUSSION TITLE
Conversion Error

### DISCUSSION THREAD

**Discussion by kill3rbytezzz** [Kas-tle/java2bedrock.sh#149] [Converter Help] [answered]
Helo

I'm trying to convert some Resource Pack from java to Bedrock RP, it is a Slimefun Resource Pack perhaps bedrock player can get a better icon and item for their Slimefun, bat after doing the convert i did'nt get any output at config.json either geyser-models.json
I think this would be my Resource Pack problem, since i saw some Resource Pack on converted well on the converter
here's my resource pack (https://dl.dropboxusercontent.com/s/nb7kmh6bbg9lav0/Slimefun%20Texture%203.1.1%20%28MC%201.16%29.zip?dl%5C=1)
any help would be appreciated

Thanks,

**Comment by typicalsmc** [comment]
link conversion issue link too please

**Reply by kill3rbytezzz** [reply]
Here's the conversion issue, i used it three times
- https://github.com/Kas-tle/java2bedrock.sh/issues/146
- https://github.com/Kas-tle/java2bedrock.sh/issues/147
- https://github.com/Kas-tle/java2bedrock.sh/issues/148
all of them failed

**Comment by Kas-tle** [comment, marked as answer]
This pack is extremely poorly formatted. For some reason, rather than simply omit entries when no model exist, the author has put hundreds of entries in each predicate file that point to "item/raulh22/slimefun/slimefun/none". Because of this, after the config is generated, there are likely thousands of useless entries that the converter must still, nonetheless, attempt to parse. This leads to the converter running up against the limit of arguments for a bash command...

Please remove these entries as they really have almost no purpose.

**Reply by kill3rbytezzz** [reply]
so i assume i'm only needed to remove just the entries that pointed to "item/raulh22/slimefun/slimefun/none" on the .json resource pack?? iil try to do that and giving the update after, thank you very much for the replies...

**Reply by Kas-tle** [reply]
Also it seems that even with these removed, there are still about 20,000 config entries... you will likely have to run the converter locally if you wish to convert this pack, as there is a 30 minute limit on action runs to prevent abuse.

**Reply by kill3rbytezzz** [reply]
i already did remove the entries that pointing to none and its making another error on resource pack, i've done the converter locally as well and it took a long time to finish, i think its the resource pack proble, thanks for your help but i think nothing i can do about this, once again thank you very much....

**Reply by Kas-tle** [reply]
Could you provide the link to the updated pack and I will see if there's anything I can do to debug?

**Reply by kill3rbytezzz** [reply]
https://cdn.discordapp.com/attachments/461422152445591562/1002268857446191176/Slimefun_Texture_3.1.1_MC_1.16.zip
here the new resource pack, i deleted the all of the entries that pointing to "item/raulh22/slimefun/slimefun/none", i create a simple python script to do this, i checked by myself aswell, all the entries mentioned already removed...

**Reply by Kas-tle** [reply]
The script you used has created invalid json in every predicate file.
![image](https://user-images.githubusercontent.com/26531652/181607178-0c28c8d0-03e9-4dae-824b-8888c911a3ad.png)

**Reply by Kas-tle** [reply]
Also is there a functional reason as to why every predicate file is identical...? What is the point is having all of these items that are model duplicates of the same item...? Does slimefun actually utilize all of these?

**Reply by kill3rbytezzz** [reply]
ah, since i'm only removing the none entries, and its probably on the last line, so i forgot to remove the coma on the last line, iil work out for this, and actually i got this pack from here https://www.planetminecraft.com/texture-pack/slimefun-texture-by-raulh22/ but i can safely assume thats slimefun using same item with different function e.g (Talisman and stuff)...

**Reply by kill3rbytezzz** [reply]
after i change the error part u mention, some of the item thats not come from vanilla minecraft is showing (custom item) but item from vanilla minecraft is glitch and not showing anything except the error block, i actually dont really understand how resource pack work and not too familiar with json, i'm so sorry i bother you with this...

**Reply by Kas-tle** [reply]
What it looks like to me is the person who made this pack was incredibly lazy and rather than actually find the proper items that slimefun uses for each custom item, they simply applied all of them to every item. The PMC link you provided has a configuration file that goes with slimefun that allows the mapping of custom model data to slimefun items. I suspect that the maker of this pack did not want to look at the slimefun source to see what custom items maps to what vanilla item, and instead opted to copy all custom model data entries for every vanilla item slimefun makes use of.

Indeed, even with your stated example of the Talisman, if we look to the slimefun source, we see that only the emerald is actually used for all its variants: https://github.com/Slimefun/Slimefun4/blob/7274436ca1e39ad1267ea4b24aa39f127fb29898/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java#L546

**Reply by kill3rbytezzz** [reply]
agreed, but after all im not good at resource pack thing so i have to used this one, probably to make thing easier i have to learn by myself how to make the resource pack itself, thanks for your help and i think there is no other way aside than creating the resource pack fom the scratch, since changing other people resource pack gonna take a lot of effort and also probably have to deal with some licence stuff. once again thank you very much for the help...
